smoke/ip_forward_frr_test.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Maxime Leroy
+
+set -e
+
+test_frr=true
+
+. $(dirname $0)/_init.sh
+
+tap_index=0
+create_interface() {
+    local p="$1"
+    local mac="$2"
+
+    grcli add interface port $p devargs net_tap$tap_index,iface=tap-$p,remote=v0-$p mac $mac
+
+    local max_tries=5
+    local count=0
+    while vtysh -c "show interface $p" 2>&1 | grep -q "% Can't find interface"; do
+	    if [ "$count" -ge "$max_tries" ]; then
+		    echo "Interface $p not found after $max_tries attempts."
+		    exit 1
+	    fi
+	    sleep 1
+	    count=$((count + 1))
+    done
+
+    tap_index=$((tap_index + 1))
+}
+
+set_ip_address() {
+    local p="$1"
+    local ip_cidr="$2"
+    local max_tries=5
+    local count=0
+
+    vtysh -c "configure terminal" \
+          -c "interface ${p}" \
+          -c "ip address ${ip_cidr}" \
+          -c "exit"
+
+    while ! grcli show ip address 2>/dev/null | grep -q "^${p}[[:space:]]\+${ip_cidr}$"; do
+	    if [ "$count" -ge "$max_tries" ]; then
+		    echo "Ip address $ip_cidr not set after $max_tries attempts."
+		    exit 1
+	    fi
+	    sleep 1
+	    count=$((count + 1))
+    done
+}
+
+set_ip_route() {
+    local prefix="$1"
+    local next_hop="$2"
+    local max_tries=5
+    local count=0
+
+    vtysh -c "configure terminal" \
+          -c "ip route ${prefix} ${next_hop}" \
+          -c "exit"
+
+    while ! grcli show ip route 2>/dev/null | grep -q "^0[[:space:]]\+${prefix}[[:space:]]\+${next_hop}[[:space:]]"; do
+	    if [ "$count" -ge "$max_tries" ]; then
+		    echo "Route ${prefix} via ${next_hop} not found after ${max_tries} attempts."
+		    exit 1
+	    fi
+	    sleep 1
+	    count=$((count + 1))
+    done
+}
+
+p0=${run_id}0
+p1=${run_id}1
+# use veth pair because tap driver doesn't support namespace
+ip link add v0-$p0 type veth peer name v1-$p0
+ip link add v0-$p1 type veth peer name v1-$p1
```

Comment:
Put this into `create_interface` maybe? And move veth interfaces into netns later.